docs/specification/draft/basic/streams.mdx

```diff
@@ -0,0 +1,240 @@
+```
+
+Example `stream/resume` request:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "method": "stream/resume",
+  "params": {
+    "streamId": "550e8400-e29b-41d4-a716-446655440000"
+  }
+}
+```
+
+When a client sends a `stream/resume` request, the server should:
+1. Send all unsent messages for the stream to the client
+2. If the stream has ended, send a `notifications/stream/end` notification after sending all other unsent messages
+3. If the stream ID is invalid (e.g., expired), respond with an error
+
+
+#### Streamable HTTP Considerations
+
+When using the Streamable HTTP transport, streams are implemented using the Server-Sent Events (SSE) protocol. When served over HTTP, a server cannot definitively when any given message is delivered. Clients use the `Last-Event-ID` HTTP header to resume their SSE connection to Streamable HTTP servers.
+
+In order to deliver durable message guarantees, servers may maintain a buffer of recently sent messages (a two minute window aligns with common TCP idle timeout configurations), and use a scheme to correlate message IDs across streams, such as by using a strictly monotonic counter. When a client reconnects after a disconnection, the `Last-Event-ID` header can be used to derive the point from which its view of streams will resume and poll from.
+
+
+### Polling
+
+Clients can check a stream's status without receiving all unsent messages by sending a `stream/poll` request:
+
+```mermaid
+sequenceDiagram
+    participant Client
+    participant Server
+
+    Client->>+Server: Request (e.g., tools/call)
+    Server-->>Client: notifications/stream/create
+    Server-->>Client: Initial messages
+
+    Note over Client,Server: Disconnection occurs
+
+    Client->>Server: stream/poll
+    Server-->>Client: StreamPollResult
+
+    Note over Client,Server: Client decides to resume
+
+    Client->>Server: stream/resume
+    Server-->>Client: Unsent messages
+    Server-->>-Client: notifications/stream/end
+```
+
+Example `stream/poll` request and response:
+
+```json
+// Request
+{
+  "jsonrpc": "2.0",
+  "id": 2,
+  "method": "stream/poll",
+  "params": {
+    "streamId": "550e8400-e29b-41d4-a716-446655440000"
+  }
+}
+
+// Response
+{
+  "jsonrpc": "2.0",
+  "id": 2,
+  "result": {
+    "streamId": "550e8400-e29b-41d4-a716-446655440000",
+    "status": "live",
+    "pendingMessages": true,
+    "hasRequest": false,
+    "hasError": false
+  }
+}
+```
+
+The response includes:
+- `status`: The stream state - "live", "completed", or "abandoned"
+- `pendingMessages`: Whether there are unsent messages
+- `hasRequest`: Whether any pending messages include a server-to-client request
+- `hasError`: Whether any pending messages include an error result
+
+### Abandonment
+
+If a client fails to resume or poll a stream within the `resumeInterval.max` period, the server may consider the stream abandoned and reclaim resources. The next time a client attempts to resume or poll the stream, the server will respond with an error.
```

Comment:
```suggestion
If a client fails to resume or poll a stream within the `resumeInterval.max` period, the server may consider the stream abandoned and reclaim resources. The next time a client attempts to resume, the server will respond with an error.
```

Depending on how aggressive the server is about reclaiming resources, it could respond to `stream/poll` with `StreamStatus.status == "abandoned"` instead of an error.

Alternatively, we could remove the `"abandoned"` option from the enum.

Reply:
I would be generally good removing `abandoned` as a state. As a client I would treat them identically.